pkg/objectio/writer.go

```diff
@@ -337,12 +338,19 @@ func (w *objectWriterV1) Sync(ctx context.Context, items ...WriteOptions) error
 }
 
 func (w *objectWriterV1) WriteWithCompress(offset uint32, buf []byte) (data []byte, extent Extent, err error) {
+	var tmpData []byte
 	dataLen := len(buf)
-	data = make([]byte, lz4.CompressBlockBound(dataLen))
-	if data, err = compress.Compress(buf, data, compress.Lz4); err != nil {
+	compressBlockBound := lz4.CompressBlockBound(dataLen)
+	if len(w.compressBuf) < compressBlockBound {
+		w.compressBuf = make([]byte, compressBlockBound)
+	}
+	if tmpData, err = compress.Compress(buf, w.compressBuf[:compressBlockBound], compress.Lz4); err != nil {
```

Comment:
i think there are 3 case:
1.  no compressed result, then you should call `NewExtent(compress.None, ...)` next
2. len(compressed data) =~ len( raw data), consider the function `copy` has extra cost.
3. len(compressed data) << len( raw data)